modules/table/src/main/java/org/apache/ignite/internal/table/distributed/TableManager.java

```diff
@@ -964,23 +967,30 @@ public void stop() {
         metaStorageMgr.unregisterWatch(stableAssignmentsRebalanceListener);
         metaStorageMgr.unregisterWatch(assignmentsSwitchRebalanceListener);
 
-        Map<Integer, TableImpl> tablesToStop = Stream.concat(latestTablesById().entrySet().stream(), pendingTables.entrySet().stream())
-                .collect(Collectors.toMap(Map.Entry::getKey, Map.Entry::getValue, (v1, v2) -> v1));
+        var tablesToStop = new HashMap<Integer, TableImpl>();
+
+        tablesToStop.putAll(latestTablesById());
+        tablesToStop.putAll(pendingTables);
 
         cleanUpTablesResources(tablesToStop);
+    }
 
-        try {
-            IgniteUtils.closeAllManually(lowWatermark, mvGc);
-        } catch (Throwable t) {
-            LOG.error("Failed to close internal components", t);
+    @Override
+    public void stop() throws Exception {
```

Comment:
I almost sure that in some tests we may just call stop instead of proper chain of beforeNodeStop + stop. I'd add corresponding assertion that beforeStopGuard is true here in order to check (and fix) the guess.

Reply:
Great idea!